test/acceptance/user.controller.acceptance.ts

```diff
@@ -135,36 +138,40 @@ describe('UserController', () => {
   });
 
   describe('authentication functions', () => {
-    // TODO: fix storing the plain password in the following issue:
-    // https://github.com/strongloop/loopback-next/issues/1996
+    let plainPassword: string;
+
+    before('create new user', async () => {
+      plainPassword = user.password;
+      // Salt + Hash Password
+      user.password = await hashAsync(user.password, 10);
```

Comment:
Should it be the responsibility of the controller/repository to hash passwords?

Reply:
@jannyHou and I were talking about a hash utility function which can be called from the user repo or controller. I think it makes sense for the repository to call it when storing user passwords in the backend, but I see how that can be delegated to a controller too.

Reply:
@raymondfeng @b-admike My opinion is unless we define a specific repository interface for model User and add a function for the hashing purpose, I tend to put the logic in controller. And as @b-admike said, we can extract it into utility/service that can be called from the controller.
Like what we have in [authentication utils](https://github.com/strongloop/loopback4-example-shopping/blob/master/src/utils/user.authentication.ts).

Reply:
And I plan to refactor them into the util in the next PR, this one focus on fixing the plain password.

Explained in the PR description :)

A follow up PR for #26
And connect to the first acceptance criteria in #35 (there will be another PR after it to refactor the hash utils into a proper place.)

Reply:
@jannyHou Please do extract this code into utility/service that can be called from multiple places. For example tests.

Reply:
@bajtos See the refactor PR https://github.com/strongloop/loopback4-example-shopping/pull/41